### PR TITLE
[ome_sesssion] Module: ome_session

### PR DESCRIPTION
**Pull request by Kritika-Bhateja-03**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

**Review by ABHISHEK-SINHA10 — APPROVED** [review]

File: plugins/modules/idrac_session.py [hunk 1]
```diff
@@ -65,10 +65,10 @@
     choices: [present, absent]
     type: str
     default: present
-  auth_token:
```
Comment (sachin-apa) [inline plugins/modules/idrac_session.py:68]: Lets add `auth_token` as alias just to keep the back war compatibility.

File: plugins/modules/ome_session.py [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+  state:
+    description:
+     - The state of the session in OpenManage Enterprise.
+     - C(present) creates a session.
+     - C(absent) deletes a session.
+     - Module will always report changes found to be applied when I(state) is C(present).
+    choices: [present, absent]
+    type: str
+    default: present
+  x_auth_token:
```
Comment (sachin-apa) [inline plugins/modules/ome_session.py:70]: Lets add `auth_token` tp keep it consistent with the idrac_session module.

File: playbooks/idrac/ome_session.yml
Comment (sachin-apa) [inline playbooks/idrac/ome_session.yml:1]: Now we are modifying the modules to accept the auth token, can we add example here on how to use this module with the other modules.

Same for idrac_session.

File: plugins/modules/idrac_session.py (hunk 1 above)
Comment (Kritika-Bhateja-03) [reply plugins/modules/idrac_session.py:68]: Done

File: playbooks/idrac/ome_session.yml
Comment (Kritika-Bhateja-03) [reply playbooks/idrac/ome_session.yml:1]: Done

File: plugins/modules/ome_session.py (hunk 2 above)
Comment (Kritika-Bhateja-03) [reply plugins/modules/ome_session.py:70]: Done

**Comment by rajshekarp87** [comment]
Update "Version" in both idrac_session.py and test_idrac_session.py file

**Review by ABHISHEK-SINHA10 — CHANGES_REQUESTED** [review]
@Kritika-Bhateja-03

Can we update AnsibleModule --> OmeAnsibleModule, and remove the same parameters like username, password, port, x_auth_token etc, common parameters from this file, as it'll help us maintain same file to update in future.

File: plugins/modules/ome_session.py [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+short_description: Manage OpenManage Enterprise and OpenManage Enterprise modular sessions
+version_added: "9.3.0"
+description:
+  - This module allows you  to create and delete sessions on OpenManage Enterprise and OpenManage Enterprise Modular.
+options:
+  hostname:
+    description:
+      - IP address or hostname of the OpenManage Enterprise.
+    type: str
+  username:
```
Comment (sachin-apa) [inline plugins/modules/ome_session.py:29]: Add Env variables for Creds, update for idrac session as well.

Comment (Kritika-Bhateja-03) [reply plugins/modules/ome_session.py:29]: done

File: playbooks/idrac/ome_session.yml
Comment (sachin-apa) [inline playbooks/idrac/ome_session.yml:1]: Update the example on how can this be used with a module.

File: plugins/module_utils/session_utils.py [hunk 4]
```diff
@@ -242,25 +245,27 @@ def _url_common_args_spec(self, method, api_timeout, headers=None):
...
             "method": method,
             "validate_certs": self.validate_certs,
             "ca_path": self.ca_path,
             "use_proxy": self.use_proxy,
             "headers": req_header,
             "timeout": api_timeout,
-            "follow_redirects": 'all',
+            "follow_redirects": 'all'
         }
+        if "redfish" not in uri:
```
Comment (sachin-apa) [inline plugins/module_utils/session_utils.py]: Please update this to take url_kwargs as arguments instead of adding conditions.

**Review by singh-lovepreet1 — DISMISSED** [review]

File: plugins/modules/idrac_session.py
Comment (sachin-apa) [inline plugins/modules/idrac_session.py:1]: Generate rst doc for the file.

**Review by rajshekarp87 — DISMISSED** [review]

File: tests/unit/plugins/modules/test_ome_session.py [hunk 5]
```diff
@@ -0,0 +1,499 @@
+# -*- coding: utf-8 -*-
+
+#
+# Dell OpenManage Ansible Modules
+# Version 9.2.0
```
Comment (ABHISHEK-SINHA10) [inline tests/unit/plugins/modules/test_ome_session.py]: Update this to 9.3.0

**Comment by ABHISHEK-SINHA10** [comment]
Please add screenshot for code-coverage, In github it's not showing.

**Comment by Kritika-Bhateja-03** [comment]
image

File: playbooks/idrac/ome_session.yml
Comment (Kritika-Bhateja-03) [reply playbooks/idrac/ome_session.yml:1]: done

File: plugins/module_utils/session_utils.py (hunk 4 above)
Comment (Kritika-Bhateja-03) [reply plugins/module_utils/session_utils.py]: done

**Review by singh-lovepreet1 — DISMISSED** [review]

File: plugins/module_utils/session_utils.py [hunk 6]
```diff
@@ -242,25 +245,29 @@ def _url_common_args_spec(self, method, api_timeout, headers=None):
...
             "headers": req_header,
             "timeout": api_timeout,
-            "follow_redirects": 'all',
+            "follow_redirects": 'all'
         }
-        return url_kwargs
+        if url_kwargs is None:
+            return url_params
+        else:
+            url_kwargs.update(url_params)
```
Comment (sachin-apa) [inline plugins/module_utils/session_utils.py]: Can we keep this simple and not modify the input param, rather add the input param to to existing map and return the same
```
if url_kwargs:
    url_params.update(url_kwargs)
return url_params
```

**Review by ABHISHEK-SINHA10 — DISMISSED** [review]

File: plugins/modules/ome_session.py [hunk 7]
```diff
@@ -0,0 +1,411 @@
...
+    description:
+      - Username of the OpenManage Enterprise. If the username is not provided, then
+        the environment variable C(OME_USERNAME) is used.
```
Comment (ABHISHEK-SINHA10) [inline plugins/modules/ome_session.py]:
```suggestion
        the environment variable E(OME_USERNAME) is used.
```

File: plugins/modules/ome_session.py [hunk 8]
```diff
@@ -0,0 +1,411 @@
...
+      - I(username) is required when I(state) is C(present).
+    type: str
+  password:
+    description:
+      - Password of the OpenManage Enterprise. If the password is not provided, then
+        the environment variable C(OME_PASSWORD) is used.
```
Comment (ABHISHEK-SINHA10) [inline plugins/modules/ome_session.py]:
```suggestion
        the environment variable E(OME_PASSWORD) is used.
```

File: plugins/modules/idrac_session.py [hunk 9]
```diff
@@ -28,12 +28,14 @@
     type: str
   username:
     description:
-      - Username of the iDRAC.
+      - Username of the iDRAC. If the username is not provided, then
+        the environment variable C(IDRAC_USERNAME) is used.
```
Comment (ABHISHEK-SINHA10) [inline plugins/modules/idrac_session.py]:
```suggestion
        the environment variable E(IDRAC_USERNAME) is used.
```

File: plugins/modules/idrac_session.py [hunk 10]
```diff
@@ -28,12 +28,14 @@
...
       - I(username) is required when I(state) is C(present).
     type: str
   password:
     description:
-      - Password of the iDRAC.
+      - Password of the iDRAC. If the password is not provided, then
+        the environment variable C(IDRAC_PASSWORD) is used.
```
Comment (ABHISHEK-SINHA10) [inline plugins/modules/idrac_session.py]:
```suggestion
        the environment variable E(IDRAC_PASSWORD) is used.
```

File: docs/modules/ome_session.rst
Comment (ABHISHEK-SINHA10) [inline docs/modules/ome_session.rst:1]: Regenerate rst for ome_session and idrac_session after addressing below comments.

**Review by rajshekarp87 — APPROVED** [review]

**Review by sachin-apa — APPROVED** [review]

**Pull request merged**